nebula_decoders/include/nebula_decoders/nebula_decoders_hesai/decoders/pandar_xt16.hpp

```diff
@@ -0,0 +1,91 @@
+// Copyright 2024 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "nebula_decoders/nebula_decoders_hesai/decoders/hesai_packet.hpp"
+#include "nebula_decoders/nebula_decoders_hesai/decoders/hesai_sensor.hpp"
+
+#include <vector>
+
+namespace nebula::drivers
+{
+
+namespace hesai_packet
+{
+
+#pragma pack(push, 1)
+
+struct TailXT16
+{
+  uint8_t reserved1[10];
+  uint8_t return_mode;
+  uint16_t motor_speed;
+  DateTime<1900> date_time;
+  uint32_t timestamp;
+  uint8_t factory_information;
+};
```

Comment:
THis is identical to the one from XT32, consider creating a common `TailXT16_32` or do a `using TailXT16 = TailXT32;` so we have less duplicate code.

Reply:
OK